### DISCUSSION TITLE
Module Aliases?

### DISCUSSION THREAD

**Discussion by arschles** [gomods/athens#1559] [General] [answered]
In https://github.com/gomods/athens/issues/1315, someone wants to implement server-side aliasing. I'm pretty sure we can't do that because if we `go get my/aliased/module` on the server but the `go.mod` for that module has `module real/name`, the operation will fail. I think the same goes for `go mod download`. I hope that I'm right about that 😆.

If a user does `go get my/aliased/module` and somehow Athens was able to package that module name up, would the user's `go get` fail? We've talked a lot about this before, but I'm having a tough time wrapping my mind around everything.

**Comment by marwan-at-work** [comment, marked as answer]
It would be nice if we can convert *that* issue into a discussion 😄 but i'll comment there for transparency

**Comment by arschles** [comment]
100% agree. Thanks @marwan-at-work !

**Comment by arschles** [comment]
Just read your [comment](https://github.com/gomods/athens/issues/1315#issuecomment-591737107), and that's I think the second time that vanity servers have come up. We might consider adding that to as a theme to https://github.com/gomods/athens/discussions/1549

**Comment by qorncat** [comment]
pls mark answer and then remove again
